### PR TITLE
Add table ec2_capacity_reservation. Closes #542

### LINKED ISSUE #542: Add table ec2_capacity_reservation
**References**
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_capacity_reservation

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
arnab@turbotindias-MacBook-Pro aws-test % ./tint.js aws_ec2_capacity_reservation
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_ec2_capacity_reservation []

PRETEST: tests/aws_ec2_capacity_reservation

TEST: tests/aws_ec2_capacity_reservation
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_ec2_capacity_reservation.named_test_resource will be created
  + resource "aws_ec2_capacity_reservation" "named_test_resource" {
      + arn                     = (known after apply)
      + availability_zone       = "us-east-1a"
      + ebs_optimized           = false
      + end_date_type           = "unlimited"
      + ephemeral_storage       = false
      + id                      = (known after apply)
      + instance_count          = 1
      + instance_match_criteria = "open"
      + instance_platform       = "Linux/UNIX"
      + instance_type           = "t2.micro"
      + owner_id                = (known after apply)
      + tags_all                = (known after apply)
      + tenancy                 = "default"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id        = "632902152528"
  + availability_zone = "us-east-1a"
  + aws_partition     = "aws"
  + instance_count    = 1
  + instance_platform = "Linux/UNIX"
  + instance_type     = "t2.micro"
  + region_name       = "us-east-1"
  + resource_aka      = (known after apply)
  + resource_id       = (known after apply)
aws_ec2_capacity_reservation.named_test_resource: Creating...
aws_ec2_capacity_reservation.named_test_resource: Creation complete after 3s [id=cr-0b93d80592d868a60]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = "632902152528"
availability_zone = "us-east-1a"
aws_partition = "aws"
instance_count = 1
instance_platform = "Linux/UNIX"
instance_type = "t2.micro"
region_name = "us-east-1"
resource_aka = "arn:aws:ec2:us-east-1:632902152528:capacity-reservation/cr-0b93d80592d868a60"
resource_id = "cr-0b93d80592d868a60"

Running SQL query: test-get-query.sql
[
  {
    "account_id": "632902152528",
    "availability_zone": "us-east-1a",
    "capacity_reservation_id": "cr-0b93d80592d868a60",
    "instance_platform": "Linux/UNIX",
    "instance_type": "t2.micro",
    "partition": "aws",
    "region": "us-east-1",
    "state": "active"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "account_id": "632902152528",
    "availability_zone": "us-east-1a",
    "capacity_reservation_id": "cr-0b93d80592d868a60",
    "instance_platform": "Linux/UNIX",
    "instance_type": "t2.micro",
    "partition": "aws",
    "region": "us-east-1",
    "state": "active"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:ec2:us-east-1:632902152528:capacity-reservation/cr-0b93d80592d868a60"
    ],
    "title": "cr-0b93d80592d868a60"
  }
]
✔ PASSED

POSTTEST: tests/aws_ec2_capacity_reservation

TEARDOWN: tests/aws_ec2_capacity_reservation

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  capacity_reservation_id,
  capacity_reservation_arn,
  instance_type,
  state
from
  aws_ec2_capacity_reservation;
+-------------------------+------------------------------------------------------------------------------+---------------+---------+
| capacity_reservation_id | capacity_reservation_arn                                                     | instance_type | state   |
+-------------------------+------------------------------------------------------------------------------+---------------+---------+
| cr-0b30935e9fc2da81e    | arn:aws:ec2:us-east-1:632902152528:capacity-reservation/cr-0b30935e9fc2da81e | t2.micro      | expired |
| cr-0e866f1c8e19c2572    | arn:aws:ec2:us-east-1:632902152528:capacity-reservation/cr-0e866f1c8e19c2572 | t2.micro      | active  |
+-------------------------+------------------------------------------------------------------------------+---------------+---------+
> select
  capacity_reservation_id,
  capacity_reservation_arn,
  instance_type,
  state
from
  aws_ec2_capacity_reservation
where
  state = 'expired';
+-------------------------+------------------------------------------------------------------------------+---------------+---------+
| capacity_reservation_id | capacity_reservation_arn                                                     | instance_type | state   |
+-------------------------+------------------------------------------------------------------------------+---------------+---------+
| cr-0b30935e9fc2da81e    | arn:aws:ec2:us-east-1:632902152528:capacity-reservation/cr-0b30935e9fc2da81e | t2.micro      | expired |
+-------------------------+------------------------------------------------------------------------------+---------------+---------+
> select
  capacity_reservation_id,
  capacity_reservation_arn,
  instance_type,
  state
from
  aws_ec2_capacity_reservation
where
  capacity_reservation_id = 'cr-0b30935e9fc2da81e';
+-------------------------+------------------------------------------------------------------------------+---------------+---------+
| capacity_reservation_id | capacity_reservation_arn                                                     | instance_type | state   |
+-------------------------+------------------------------------------------------------------------------+---------------+---------+
| cr-0b30935e9fc2da81e    | arn:aws:ec2:us-east-1:632902152528:capacity-reservation/cr-0b30935e9fc2da81e | t2.micro      | expired |
+-------------------------+------------------------------------------------------------------------------+---------------+---------+
> 
```
</details>

### CODE REVIEW

File: aws-test/tests/aws_ec2_capacity_reservation/variables.tf [hunk 1]
```diff
@@ -0,0 +1,85 @@
+
```
Comment (bigdatasourav) [inline aws-test/tests/aws_ec2_capacity_reservation/variables.tf:1]:
```suggestion
```

File: aws/table_aws_ec2_capacity_reservation.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsEc2CapacityReservation(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_ec2_capacity_reservation",
+		Description: "AWS EC2 capacity reservation",
```
Comment (bigdatasourav) [inline aws/table_aws_ec2_capacity_reservation.go:19]:
```suggestion
		Description: "AWS EC2 Capacity Reservation",
```

File: aws/table_aws_ec2_capacity_reservation.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+			Hydrate:           getEc2CapacityReservation,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listEc2CapacityReservations,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "capacity_reservation_arn",
+				Description: "The Amazon Resource Name (ARN) of the capacity reservation.",
```
Comment (bigdatasourav) [inline aws/table_aws_ec2_capacity_reservation.go:32]: Please add arn column after ID

File: aws/table_aws_ec2_capacity_reservation.go [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+		return nil, err
+	}
+
+	if op.CapacityReservations != nil && len(op.CapacityReservations) > 0 {
+		return op.CapacityReservations[0], nil
+	}
+	return nil, nil
+}
+
+func getAwsEc2CapacityReservationAkas(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (bigdatasourav) [inline aws/table_aws_ec2_capacity_reservation.go:200]: we dont need this as we are getting arn in api response

File: aws/table_aws_ec2_capacity_reservation.go [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Tags"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsEc2CapacityReservationAkas,
```
Comment (bigdatasourav) [inline aws/table_aws_ec2_capacity_reservation.go:138]:
```suggestion
```

File: aws/table_aws_ec2_capacity_reservation.go [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+				Transform:   transform.FromValue().Transform(transform.EnsureStringArray),
```
Comment (bigdatasourav) [inline aws/table_aws_ec2_capacity_reservation.go:139]: use capacity_reservation_arn

File: aws/table_aws_ec2_capacity_reservation.go [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+			{
+				Name:        "tenancy",
+				Description: "Indicates the tenancy of the capacity reservation. A capacity reservation can have one of the following tenancy settings: 'default', 'dedicated'.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "total_instance_count",
+				Description: "The total number of instances for which the capacity reservation reserves capacity",
+				Type:        proto.ColumnType_INT,
+			},
```
Comment (bigdatasourav) [inline aws/table_aws_ec2_capacity_reservation.go:119]: Add tag_src column

File: aws/table_aws_ec2_capacity_reservation.go [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CapacityReservationId"),
+			},
+			{
+				Name:        "tags",
```
Comment (bigdatasourav) [inline aws/table_aws_ec2_capacity_reservation.go:129]: Format tags as map[string]string{}, For reference check aws_tagging_resource table.

**Review by bigdatasourav — CHANGES_REQUESTED** [review]
@c0d3r-arnab Please check the requested changes also add one get call example in example query

File: aws/table_aws_ec2_capacity_reservation.go [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "tenancy",
+				Description: "Indicates the tenancy of the capacity reservation. A capacity reservation can have one of the following tenancy settings: 'default', 'dedicated'.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "total_instance_count",
+				Description: "The total number of instances for which the capacity reservation reserves capacity",
```
Comment (bigdatasourav) [inline aws/table_aws_ec2_capacity_reservation.go:117]:
```suggestion
				Description: "The total number of instances for which the capacity reservation reserves capacity.",
```

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**